examples/secured-client-with-basic-auth/secured_client_with_basic_auth.bal

```diff
@@ -5,33 +5,30 @@ import ballerina/log;
 // Define the basic auth client endpoint to call the backend services.
 // Basic authentication is enabled by setting the `scheme: http:BASIC_AUTH`
 // The `username` and `password` should be specified as needed.
-endpoint http:Client httpEndpoint {
-    url: "https://localhost:9090",
+http:Client httpEndpoint = new("https://localhost:9090", config = {
     auth: {
         scheme: http:BASIC_AUTH,
         username: "tom",
         password: "1234"
     }
-};
+});
 
 public function main() {
     // This defines the authentication credentials of the HTTP service.
     config:setConfig("b7a.users.tom.password", "1234");
 
     // Send a `GET` request to the specified endpoint.
     var response = httpEndpoint->get("/hello/sayHello");
-    match response {
-        http:Response resp => {
-            var result = resp.getPayloadAsString();
-            log:printInfo(result is error ? "Failed to retrieve payload." : result);
-        }
-        error err => log:printError("Failed to call the endpoint.");
+    if (response is http:Response) {
+        var result = response.getPayloadAsString();
+        log:printInfo((result is error) ? "Failed to retrieve payload." : result);
+    } else {
+        log:printError("Failed to call the endpoint.");
```

Comment:
Shall we print the actual error as well that result from get method.

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/11969/commits/ac0fe16ec49c2ab56e5e39177aa86d4e6a352942